### PR TITLE
Reorganize packages and modules

### PR DESCRIPTION
**Pull request by jingtang10**
- Change namespace com.google.fhirengine to com.google.android.fhir
- Rename module fhirengine to core
- Rename module example to reference

### CODE REVIEW

File: core/src/main/java/com/google/android/fhir/db/impl/DatabaseImpl.kt [hunk 1]
```diff
@@ -14,30 +14,30 @@
...
+import com.google.android.fhir.index.FhirIndexer
+import com.google.android.fhir.resource.getResourceType
+import com.google.android.fhir.search.impl.Query
 import org.hl7.fhir.r4.model.Resource
 import org.hl7.fhir.r4.model.ResourceType
 
 /**
  * The implementation for the persistence layer using Room.
- * See docs for [com.google.fhirengine.db.Database] for the API docs.
+ * See docs for [com.google.core.db.Database] for the API docs.
```
Comment (deepankarb) [inline core/src/main/java/com/google/android/fhir/db/impl/DatabaseImpl.kt:33]: Change to `com.google.android.fhir.db.Database`.

**Review by deepankarb — COMMENTED** [review]
Approved with a nit!

File: core/src/main/java/com/google/android/fhir/db/impl/DatabaseImpl.kt (hunk 1 above)
Comment (jingtang10) [reply core/src/main/java/com/google/android/fhir/db/impl/DatabaseImpl.kt]: done!

**Comment by jingtang10** [comment]
Thanks @deepankarb. Addressed your comment. I think I still need an approval before I can merge :) thanks!

**Review by pankajkatoch — APPROVED** [review]

**Review by yigit — COMMENTED** [review]
lgtm

**Pull request merged**